checker/src/main/java/org/checkerframework/checker/index/upperbound/OffsetEquation.java

```diff
@@ -266,6 +274,13 @@ private Integer evalConstantTerm(Receiver termReceiver) {
                     }
                 }
             }
+        } else if (factory != null && termReceiver instanceof FlowExpressions.LocalVariable) {
+            Element element = ((FlowExpressions.LocalVariable) termReceiver).getElement();
+            AnnotationMirror am =
+                    factory.getTypeFactoryOfSubchecker(ValueChecker.class)
+                            .getAnnotatedType(element)
+                            .getAnnotation(IntVal.class);
+            return IndexUtil.getIntegerFromIntValWithOneArgument(am);
         }
```

Comment:
My comment was
>You should create a version of `IndexUtil#getExactValue` that takes an element and call that method.

But that’s not what you did. Was there a reason that didn’t or won’t work?

Reply:
The method I thought I should create is a static one that takes an `AnnotationMirror` and returns an integer representing its only `IntVal` argument. If the annotation is not `IntVal` or it has more than one argument, then it returns `null`. I think I misunderstood it.

Did you mean I need to create a method that takes an `Element` and returns its `IntVal` annotation?

Reply:
I mean a method like:
```
 public static Long getExactValue(Element element, ValueAnnotatedTypeFactory factory) {
        AnnotatedTypeMirror valueType = factory.getAnnotatedType(element);
        Range possibleValues = getPossibleValues(valueType, factory);
        if (possibleValues != null && possibleValues.from == possibleValues.to) {
            return possibleValues.from;
        } else {
            return null;
        }
    }
```

Then call it:
```
Element element = ((FlowExpressions.LocalVariable) termReceiver).getElement();
Long l = getExactValue(element, factory.getTypeFactoryOfSubchecker(ValueChecker.class))
```
`l` might be null, so you'll need to deal with that case.

Reply:
Thank you for the suggestion. I updated the class accordingly.